### DISCUSSION TITLE
Setting xterm "screenReaderMode": true" disables xterm-addon-web-links?

### DISCUSSION THREAD

**Discussion by RahjIII** [sourcelair/xterm.js#4429] [Q&A] [answered]
I have an xterm.js instance running with xterm-addon-web-links enabled.  When the xterm screenReaderMode is set to false, http:// uris are detected on cursor hover, and are clickable, which is the behavior I'm looking for.

When I change the theme's screenReaderMode to true to add aria hints for screen reader software, the http:// uris are no longer detected on rollover and are no longer clickable.   Not the behavior I am looking for.  

Is this an intentional behavior for screenReaderMode?   Is there something else I should be doing to enable both xterm-addon-web-links and screenReaderMode: true at the same time?

(Chrome 110.0.5481.77, Firefox 102.7.0esr.  xterm 5.1.0, xterm-addon-web-links 0.8.0.)

Thanks for any advice!

**Comment by jerch** [comment]
Seems https://github.com/xtermjs/xterm.js/blob/4037b943b9d9b2b44ef9ab96dd09b42363c96c93/css/xterm.css#L150 is causing this. @Tyriar Is there a reason for this element having a higher z-index than the viewport div?

**Reply by RahjIII** [reply]
It looks like I can lower the z-index to make the links work,  but i don't want to break something unknown in accessibility by doing so.  My ability to test the accessibility mode is pretty limited... any advice here would help, @Tyriar !

**Reply by isubasin** [reply]
Hello @Tyriar or anyone, is there any update on this request? We beleive this is the solution for the discussion https://github.com/xtermjs/xterm.js/discussions/4425 as well.

**Reply by Tyriar** [reply]
If you lower the z index it would break reading the content under the cursor for low vision users. So currently you would need to choose between those 2 options

**Reply by RahjIII** [reply]
Ok, well that's a bummer.   I was hoping to be able to leave screenReaderMode permanently enabled, but I can't leave it on by default if it disables clickable links.  Enabling accessibility will have to be an app preference that I add sometime. 

Thanks for the answers @jerch @Tyriar , and thanks for maintaining xterm.js!  I really appreciate the work you do.

**Comment by RahjIII** [comment, marked as answer]
Verified the fix in 5.2.1.   Links are clickable regardless of screenReaderMode setting.
